PyMca5/PyMcaGui/io/TiledDataChannelTable.py

```diff
@@ -0,0 +1,191 @@
+        """
+        self.dataChannelList = channelList
+        n = len(channelList)
+        self.setRowCount(n)
+        if n > 0:
+            for i in range(n):
+                self._addLine(i, channelList[i])
+
+    def _addLine(self, i, channelLabel):
+        """
+        Adds individual line to Data Channel Table based on the label of the
+        Data Channel in the scan file.
+        """
+        # Data Channel name
+        item = self.item(i, 0)
+        if item is None:
+            item = qt.QTableWidgetItem(channelLabel)
+            item.setTextAlignment(qt.Qt.AlignHCenter | qt.Qt.AlignVCenter)
+            self.setItem(i, 0, item)
+        else:
+            item.setText(channelLabel)
+        
+        # Data Channel name is not selectable
+        item.setFlags(qt.Qt.ItemIsEnabled)
+
+        # Checkboxes
+        for j in range(1, 4):
+            widget = self.cellWidget(i, j)
+            if widget is None:
+                widget = CheckBoxItem(self, i, j)
+                self.setCellWidget(i, j, widget)
+                widget.sigCheckBoxItemSignal.connect(self._mySlot)
+            else:
+                pass
+
+            # Resize columns to fit contents
+        self.resizeColumnsToContents()
+
+        # Stretch Columns to fill the table
+        column_count = self.columnCount()
+        for column in range(column_count):
+            self.horizontalHeader().setStretchLastSection(True)
+            self.horizontalHeader().setSectionResizeMode(column, QtWidgets.QHeaderView.Stretch)
+
+    def _mySlot(self, ddict):
+        row = ddict["row"]
+        col = ddict["col"]
+        
+        # 'x' column
+        if col == 1:
+            if ddict["state"]:
+                if row not in self.xSelection:
+                    self.xSelection.append(row)
+            else:
+                if row in self.xSelection:
+                    del self.xSelection[self.xSelection.index(row)]
+        # 'y' column
+        if col == 2:
+            if ddict["state"]:
+                if row not in self.ySelection:
+                    self.ySelection.append(row)
+            else:
+                if row in self.ySelection:
+                    del self.ySelection[self.ySelection.index(row)]
+
+        # 'Mon' column
+        if col == 3:
+            if ddict["state"]:
+                if row not in self.monSelection:
+                    self.monSelection.append(row)
+            else:
+                if row in self.monSelection:
+                    del self.monSelection[self.monSelection.index(row)]
+
+        self._update()
+
+    def _update(self):
+        for i in range(self.rowCount()):
+            
+            # 'x' column
+            j = 1
+            widget = self.cellWidget(i, j)
+            if i in self.xSelection:
+                if not widget.isChecked():
+                    widget.setChecked(True)
+            else:
+                if widget.isChecked():
+                    widget.setChecked(False)
```

Comment:
Possible simplification... see whether you find this more or less clear.

```suggestion
            widget.setChecked(
                i in self.xSelection
                and not widget.isChecked()
            )
```